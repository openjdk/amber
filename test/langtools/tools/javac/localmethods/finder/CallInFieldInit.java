/**
 * @test /nodynamiccopyright/
 * @bug 8234800
 * @summary Finder crashes when encountering private method calls in field initializer
 * @compile/fail/ref=CallInFieldInit.out -Werror -XDrawDiagnostics -XDfind=local-methods CallInFieldInit.java
 */

public class CallInFieldInit {

    int x = getInt();
    int y = getAnotherInt();

    private int getInt() { return 0; }
    private int getAnotherInt() { return 0; }

    private int getIntLocalizable() { return 0; }

    void foo() {
        getIntLocalizable();
        getAnotherInt();
    }
}
