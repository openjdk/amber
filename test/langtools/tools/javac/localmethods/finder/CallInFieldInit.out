CallInFieldInit.java:16:17: compiler.warn.method.could.be.declared.local: foo(), kindname.class, CallInFieldInit
- compiler.err.warnings.and.werror
1 error
1 warning
